lib/src/glossary/widgets/filter_by.dart

```diff
@@ -0,0 +1,269 @@
+
+class FilterBy extends StatefulWidget {
+  final void Function() filterGlossary;
+  final List<JLPTLevel> selectedJlptLevel;
+  final List<KnowledgeLevel> selectedKnowledgeLevel;
+
+  const FilterBy({
+    super.key,
+    required this.filterGlossary,
+    required this.selectedJlptLevel,
+    required this.selectedKnowledgeLevel,
+  });
+
+  @override
+  State<FilterBy> createState() => _FilterBy();
+}
+
+class _FilterBy extends State<FilterBy> {
+  late List<JLPTLevel> _selectedJlptLevel;
+  late List<KnowledgeLevel> _selectedKnowledgeLevel;
+
+  @override
+  void initState() {
+    super.initState();
+    _selectedJlptLevel = widget.selectedJlptLevel;
+    _selectedKnowledgeLevel = widget.selectedKnowledgeLevel;
+  }
+
+  void _toggleJLPTLevel(JLPTLevel jlptLevel) {
+    if (_selectedJlptLevel.contains(jlptLevel)) {
+      _selectedJlptLevel.remove(jlptLevel);
+    } else {
+      _selectedJlptLevel.add(jlptLevel);
+    }
+    setState(() {
+      _selectedJlptLevel = _selectedJlptLevel;
+    });
+  }
+
+  void _toggleKnowledgeLevel(KnowledgeLevel knowledgeLevel) {
+    if (_selectedKnowledgeLevel.contains(knowledgeLevel)) {
+      _selectedKnowledgeLevel.remove(knowledgeLevel);
+    } else {
+      _selectedKnowledgeLevel.add(knowledgeLevel);
+    }
+    setState(() {
+      _selectedKnowledgeLevel = _selectedKnowledgeLevel;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations l10n = AppLocalizations.of(context);
+
+    const titlePadding = EdgeInsets.only(left: 20, top: 8, right: 20);
+    const checkboxPadding = EdgeInsets.symmetric(horizontal: 40);
+
+    const styleTitle = TextStyle(fontSize: 20, fontWeight: FontWeight.bold);
+
+    return Scaffold(
+      appBar: AppBar(
+        leading: IconButton(
+          icon: const Icon(Icons.close),
+          onPressed: () {
+            Navigator.of(context).pop();
+            widget.filterGlossary();
+          },
+        ),
+        title: Text(l10n.glossary_filter_by_title),
+      ),
+      body: Column(
+        children: [
+          Row(children: [
+            Padding(
+              padding: titlePadding,
+              child: Text(
+                l10n.glossary_filter_by_jlpt_level_title,
+                textAlign: TextAlign.left,
+                style: styleTitle,
+              ),
+            ),
+          ]),
+          Row(
+            children: [
+              Expanded(
+                flex: 9,
+                child: Padding(
+                  padding: checkboxPadding,
+                  child: Text(l10n
+                      .glossary_filter_by_jlpt_level(JLPTLevel.level1.value)),
+                ),
+              ),
+              Expanded(
+                flex: 1,
+                child: Checkbox(
+                    value: _selectedJlptLevel.contains(JLPTLevel.level1),
+                    onChanged: (_) => {_toggleJLPTLevel(JLPTLevel.level1)}),
+              ),
+            ],
+          ),
```

Comment:
Tu pourrais mettre une boucle for ici plutôt que de repeter le code pour chaque level de JLPT, pareil pour le knowledge level

Reply:
Fait